src/FastExpressionCompiler.LightExpression/Expression.cs

```diff
@@ -180,8 +180,8 @@ public static MemberExpression MakeMemberAccess(Expression expression, MemberInf
         public static UnaryExpression Not(Expression operand) =>
             new UnaryExpression(ExpressionType.Not, operand, operand.Type);
 
-        public static TypeBinaryExpression TypeAs(Expression operand, Type type) =>
-            new TypeBinaryExpression(ExpressionType.TypeAs, operand, type);

```

Comment:
Looks good, but does `TypeBinaryExpression` is used anywhere now. If not, we can remove it.

Reply:
It's used for `TypeIs` and for `TypeEqual`
I've fixed the same error with TypeIs and added support for TypeEqual - happy to move the latter into a separate PR.

If you like, I'd be happy to create another PR that adds unit tests for the different XxxExpression classes to make sure that the API matches the framework one?

Reply:
I will merge this PR and you can tackle the other cases separately - good idea.